autoload/lsp/ui/vim/completion.vim

```diff
@@ -0,0 +1,259 @@
+" vint: -ProhibitUnusedVariable
+"
+let s:context = {}
+
+function! lsp#ui#vim#completion#_setup() abort
+  augroup lsp_ui_vim_completion
+    autocmd!
+    autocmd CompleteDone * call s:on_complete_done()
+  augroup END
+endfunction
+
+"
+" After CompleteDone, v:complete_item's word has been inserted into the line.
+" Yet not inserted commit characters.
+"
+" below example uses | as cursor position.
+"
+" 1. `call getbuf|`<C-x><C-o>
+" 2. select `getbufline` item.
+" 3. Insert commit characters. e.g. `(`
+" 4. fire CompleteDone, then the line is `call getbufline|`
+" 5. call feedkeys to call `s:on_complete_done_after`
+" 6. then the line is `call getbufline(|` in `s:on_complete_done_after`
+"
+function! s:on_complete_done() abort
+  if empty(v:completed_item)
+    doautocmd User lsp_complete_done
+    return
+  endif
+
+  " Try to get managed user_data.
+  let l:managed_user_data = lsp#omni#get_managed_user_data_from_completed_item(v:completed_item)
+
+  " Clear managed user_data.
+  call lsp#omni#_clear_managed_user_data_map()
+
+  " If managed user_data does not exists, skip it.
+  if empty(l:managed_user_data)
+    doautocmd User lsp_complete_done
+    return
+  endif
+
+  let s:context['line'] = getline('.')
+  let s:context['position'] = getpos('.')
+  let s:context['completed_item'] = copy(v:completed_item)
+  let s:context['server_name'] = l:managed_user_data['server_name']
+  let s:context['completion_item'] = l:managed_user_data['completion_item']
+  call feedkeys(printf("\<C-r>=<SNR>%d_on_complete_done_after()\<CR>", s:SID()), 'n')
```

Comment:
Can I silence this line? This line echos `=<SNR>105_on_complete_done_after()`.

Reply:
I have the same opinion too. But I couldn't find the solution for it.

Reply:
If we use the timer instead of the feedkeys for delaying to the next tick, will display incomplete line to the user...

Reply:
Solved in #691.